src/extensions/data_array.js

```diff
@@ -65,6 +86,76 @@ export default () => {
           });
         }
       });
+
+      // 5) Validate that `dataArray@iot.count` and `body.dataArray.length`
+      //    are the same.
+      if (body[dataArrayIotCount] !== body.dataArray.length) {
+        throw Object.create({
+          name: ERR.BAD_REQUEST,
+          errno: ERR.ERRNO_BAD_REQUEST,
+          errors: '\'dataArray@iot.count\' doesn\'t match \'dataArray\'' +
+                  ' length.'
+        });
+      }
+    };
+
+    const createInstances = observationObjects => {
+      return db().then(models => {
+        return Promise.all(observationObjects.map(observationObj => {
+          return models.createInstance(
+            observations, { body: observationObj }).then(instance => {
+            return Promise.resolve(instance.id);
+          }).catch(() => {
+            return Promise.resolve(null);
+          });
+        }));
+      }).catch(() => {
+        return ERR.ApiError(res, 500,
+          ERR.ERRNO_INTERNAL_ERROR, ERR.INTERNAL_ERROR);
+      });
+    };
+
+    const createObservations = body => {
+      let observationObjects = [];
+      const resultIndex = body.components.indexOf(result);
+      const featureOfInterestIndex =
+        body.components.indexOf(featureOfInterestId);
+      const phenomenonTimeIndex = body.components.indexOf(phenomenonTime);
+      const dataArrayCount = body[dataArrayIotCount];
+
+      for (let i = 0; i < dataArrayCount; i++) {
+        let observation = {};
+        observation.Datastream = {};
+        observation.Datastream[iotId] = body.Datastream[iotId];
+        observation.result = body.dataArray[i][resultIndex];
+        observation.phenomenonTime = body.dataArray[i][phenomenonTimeIndex];
+
+        // If FeatureOfInterest relation is not specified,
+        // it will be extracted from Locations (i.e.,
+        // Datastreams-->Things-->Locations).
```

Comment:
Just for posterity, as mentioned on IRC:
```ssh
there are some missing association. Probably because you are creating the Datastream directly with sequelize
instead of doing `models[datastreams].create(DatastreamsRussEntity)` try `server.post('/Datastreams').send(DatastreamsRussEntity)`
this way you'll let the API handle all the creation of the associations
```